internal/config.go

```diff
@@ -0,0 +1,23 @@
+package internal
+
+import "time"
+
+// LoggingInterval returns the interval for periodic logging.
+func LoggingInterval() time.Duration {
+	return c.LoggingInterval
+}
+
+// SetLoggingInterval configures the interval for periodic logging.
+func SetLoggingInterval(i time.Duration) {
+	c.LoggingInterval = i
+}
+
+var c *config
+
+type config struct {
+	LoggingInterval time.Duration
+}
```

Comment:
This struct is used for a variable at package level that is set in `init()`. I don't think it makes sense to reuse something we have as I only want to keep information here that is needed throughout the entire Icinga DB runtime.

Reply:
My thinking behind this comment was that if we start using global variables for configuration, we possibly also avoid passing around some pointers (mainly Redis/MySQL extra options).

Possible alternative would be to store this within our `Logging` struct, but that would require more changes as we only pass `*zap.SugaredLogger` around at the moment.

Reply:
> Possible alternative would be to store this within our `Logging` struct, but that would require more changes as we only pass `*zap.SugaredLogger` around at the moment.

I have a commit lying around that does just that: introducing a custom logger with an option for the periodic logging interval. Do you want to do this instead?

Reply:
Can you push it into another branch so that I can have a look at it first?

Reply:
I just pushed it here. It's the last commit.

Reply:
I prefer the new version passing around the `*logging.Logger`, just needs some additional fixes in the delta tests.

Reply:
Sure, I'll fix the tests.

Reply:
I fixed the test and used a periodic logging interval of 1 second for our custom logger. Does that make sense or should I increase this?

Reply:
If I haven't missed anything, the delta implementation does no periodic logging, so the value doesn't matter. And if it did, 1s would probably be fine (usually, tests should be faster, but if they aren't maybe this would give a hint what's going wrong quite quickly).

Reply:
So, ready for another round of review then?